lib/ConvoyFollower/src/App.cpp

```diff
@@ -458,30 +468,43 @@ void App::processV2VCommunication()
 
     v2vStatus = m_v2vCommManager.process(vehicleStatus);
 
-    switch (v2vStatus)
+    if (m_lastV2VStatus != v2vStatus)
     {
-    case V2VCommManager::V2V_STATUS_OK:
-        /* All good. Nothing to do. */
-        break;
-
-    case V2VCommManager::V2V_STATUS_NOT_INIT:
-        LOG_WARNING("V2V not initialized.");
-        break;
-
-    case V2VCommManager::V2V_STATUS_LOST_FOLLOWER:
-    case V2VCommManager::V2V_STATUS_FOLLOWER_ERROR:
-    case V2VCommManager::V2V_STATUS_OLD_WAYPOINT:
-        LOG_ERROR("Follower Error");
-        setErrorState();
-        break;
+        /* Save last status. */
+        m_lastV2VStatus = v2vStatus;
 
-    case V2VCommManager::V2V_STATUS_GENERAL_ERROR:
-        LOG_ERROR("V2V Communication error.");
-        setErrorState();
-        break;
+        /* Process V2V status. */
+        switch (v2vStatus)
+        {
+        case V2VCommManager::V2V_STATUS_OK:
+            /* All good. Nothing to do. */
+            break;
+
+        case V2VCommManager::V2V_STATUS_NOT_INIT:
+            LOG_WARNING("V2V not initialized.");
+            break;
+
+        case V2VCommManager::V2V_STATUS_LOST_FOLLOWER:
+            /* Fallthrough */
+        case V2VCommManager::V2V_STATUS_FOLLOWER_ERROR:
+            /* Fallthrough */
+        case V2VCommManager::V2V_STATUS_OLD_WAYPOINT:
+            LOG_ERROR("Follower Error");
+            setErrorState();
+            break;
 
-    default:
-        break;
+        case V2VCommManager::V2V_STATUS_EMERGENCY:
+            setErrorState();

```

Comment:
What is the difference between error and emergency?

Reply:
Error is unexpected error. Emergency is purposefully triggered.

Reply:
Shouldn't it be handled then different?